nixos/modules/services/monitoring/beszel-agent.nix

```diff
@@ -0,0 +1,89 @@
+{
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.beszel.agent;
+in
+{
+  meta.maintainers = with lib.maintainers; [
+    BonusPlay
+    arunoruto
+  ];
+
+  options.services.beszel.agent = {
+    enable = lib.mkEnableOption "beszel agent";
+
+    package = lib.mkPackageOption pkgs "beszel" { };
+
+    environment = lib.mkOption {
+      type = lib.types.attrs;
+      default = { };
+      description = ''
+        Environment variables for configuring the beszel-agent service.
+        This field will end up public in /nix/store, for secret values use `environmentFile`.
+      '';
+    };
+    environmentFile = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+      description = ''
+        File path containing environment variables for configuring the beszel-agent service in the format of an EnvironmentFile. See {manpage}`systemd.exec(5)`.
+      '';
+    };
+    extraPath = lib.mkOption {
+      type = lib.types.listOf lib.types.package;
+      default = [ ];
+      description = ''
+        Extra packages to add to beszel path (such as nvidia-smi or rocm-smi).
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.beszel-agent = {
+      description = "Beszel Agent";
+
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+      after = [ "network-online.target" ];
+
+      environment = cfg.environment;
+      path = cfg.extraPath;
+
+      serviceConfig = {
+        ExecStart = ''
+          ${cfg.package}/bin/beszel-agent
+        '';
+
+        EnvironmentFile = cfg.environmentFile;
+
+        DynamicUser = true;
+        LockPersonality = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
```

Comment:
This causes some problems with running `nvidia-smi` with the service. I always get, that nvidia-smi can't find GPUs on the system. I guess it can be set to false? Or set it to false, if the GPU environmental variable is turned on!

Reply:
Hmm, it could be a trick if some users decide to use only `environmentFile` (which I don't think we can handle). I'm also hesitant to create dedicated `gpu` switch (but maybe that is the way?). WDYT

Reply:
I think it would make sense to turn PrivateDevices off, as it clearly seems to breaks a feature

Reply:
> Hmm, it could be a trick if some users decide to use only `environmentFile` (which I don't think we can handle). I'm also hesitant to create dedicated `gpu` switch (but maybe that is the way?). WDYT

The options are:
1. Remove the option if it isn't critical. Maybe you can explain why you added it since it wasn't present in the default systemd config provided by beszel.
2. Remove the env file option and only provide the env option. Look if the GPU attribute is provided, and if it is, check if it is `true` or `"true"`. This is much more complicated, but if the option is critical, we could keep it :)

Reply:
Just copy&paste of other modules as templates. I will remove it.